shell/ptty_unix.go

```diff
@@ -0,0 +1,82 @@
+// +build !windows
+
+package shell
+
+import (
+	"io"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+
+	"github.com/creack/pty"
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+)
+
+func runCommandWithPTTY(terragruntOptions *options.TerragruntOptions, cmd *exec.Cmd, cmdStdout io.Writer, cmdStderr io.Writer) (err error) {
```

Comment:
Did you come up with this code yourself or are you following some sort of guide or example? It seems very complicated!

Reply:
Ah I forgot to add the links. This is based on the quickstart example for https://github.com/creack/pty

Reply:
UPDATE (2704efe): added link to project and mentioned that the code is based on quickstart example.

Reply:
Ah, got it, thanks!